website/routes.py

```diff
@@ -269,6 +269,12 @@ def ember_app(path=None):
         if request.path.strip('/').startswith(k):
             ember_app = EXTERNAL_EMBER_APPS[k]
             if k == 'preprints':
+                path_values = path.split('/')
+                provider_str, guid_str = path_values[0], path_values[1]
+                preprint = Preprint.load(guid_str)
+                path = f'{provider_str}/{preprint._id}/'
+                if preprint._id != guid_str:
+                    return redirect(preprint._id, code=302)
```

Comment:
I ran to several issues: sometimes path is None, somethings the split result is None, sometimes versioned guid not found. I think we need to handle several cases + possible None result and exceptions

* osf.io/preprints/
* osf.io/prerpints/osf
* osf.io/preprints/osf/<guid>
* maybe more?

In addition, for backward compatibility, if we ran into any issue (not found or exceptions), we should fallback to the old way.